libsolidity/lsp/LanguageServer.cpp

```diff
@@ -173,6 +196,28 @@ void LanguageServer::changeConfiguration(Json::Value const& _settings)
 	}
 }
 
+vector<boost::filesystem::path> LanguageServer::allSolidityFilesFromProject() const
+{
+	namespace fs = boost::filesystem;
+
+	std::vector<fs::path> collectedPaths{};
+
+	// We explicitly decided against including all files from include paths but leave the possibility
+	// open for a future PR to enable such a feature to be optionally enabled (default disabled).
+	auto iterator = fs::recursive_directory_iterator(m_fileRepository.basePath(), fs::symlink_option::recurse);
+	auto const iteratorEnd = fs::recursive_directory_iterator();
+
+	while (iterator != iteratorEnd)
+	{
+		if (iterator->path().extension() == ".sol")
+			collectedPaths.push_back(iterator->path());
+
+		++iterator;
+	}
```

Comment:
I think that this iterator works with a range loop:
```suggestion
    auto directoryIterator = fs::recursive_directory_iterator(m_fileRepository.basePath(), fs::symlink_option::recurse);
	for (fs::directory_entry const& dirEntry: directoryIterator)
		if (dirEntry.path().extension() == ".sol")
			collectedPaths.push_back(dirEntry.path());
```